### ISSUE TITLE
Meridiem doesn't read it's value on screen readers.

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1840]
# Bug Report

## Description
The time input doesn't have aria labels on the meridiem, making it hard for screen readers to accurately interpret that screen. We need to readjust the labels so that it's clear to the user what they are inputting.

## Additional Context / Screenshots
This won't be easy possibly. When a user focuses on the meridiem input, it auto focuses to the select box, which could throw off what the screen reader is interpreting.
